cmd/delete/key/key.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2021 IBM Corp
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package key
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
+
+	"github.com/ppc64le-cloud/pvsadm/pkg"
+	"github.com/ppc64le-cloud/pvsadm/pkg/client"
+	"github.com/ppc64le-cloud/pvsadm/pkg/utils"
+)
+
+const deletePromptMessage = "Deleting all the above ssh key/key's, Do you really want to continue?"
+
+var searchStr string
+
+var Cmd = &cobra.Command{
+	Use:   "key",
+	Short: "Delete PowerVS ssh key/keys",
+	Long: `Delete PowerVS ssh key/keys
+
+# Set the API key or feed the --api-key commandline argument
+export IBMCLOUD_API_KEY=<IBM_CLOUD_API_KEY>
+
+Examples:
+
+# Delete all the ssh keys which are created before 12hrs
+pvsadm delete key --instance-name upstream-core --before 12h
+
+#Delete all the ssh keys created since 30m
+pvsadm delete key --instance-name upstream-core --since 30m
+
+#Delete all ssh keys starts with rdr-
+pvsadm delete key --instance-name upstream-core --regex "^rdr-.*"
+
+#Delete all ssh keys without asking confirmation
+pvsadm delete key --instance-name upstream-core --noprompt
+
+#Delete all ssh keys and ignore if any errors during the delete operation
+pvsadm delete key --instance-name upstream-core --ignore-errors
```

Comment:
keys api is not tied with any instance, hence wondering if we really need the instance-name or instance-id?

Reply:
From IBM UI, we can check ssh_keys irrespective of the instances, but powervs sdk needs instance id for getting the ssh keys.. 
`c.session.Power.PCloudTenantsSSHKeys.PcloudTenantsSshkeysGetall(params, ibmpisession.NewAuth(c.session, c.instanceID))`

PcloudTenantsSshkeysGetall function needs information about instanceid, hence I have added the instance-name

Reply:
can we send an empty or some dummy value there? because API doesn't I feel.